sklearn/metrics/scorer.py

```diff
@@ -290,7 +307,7 @@ def make_scorer(score_func, greater_is_better=True, needs_proba=False,
     Examples
     --------
     >>> from sklearn.metrics import fbeta_score, make_scorer
-    >>> ftwo_scorer = make_scorer(fbeta_score, beta=2)
+    >>> ftwo_scorer = make_scorer(fbeta_score, name='fbeta_score', beta=2)
```

Comment:
I think previously, @raghavrv did use `__name__` from a callable, but our consensus was that some callables won't have a `__name__` attribute.

My suggestion would be to have `name=None`, by default. And then use `hasattr(callable, "__name__")` to determine whether `callable.__name__` should be used as the shown name, or `custom_score`. (And let users override `callable.__name__` simply by passing in the name of their choice.)